src/Features/Core/Portable/Completion/Providers/SymbolCompletionItem.cs

```diff
@@ -214,22 +216,28 @@ public static SupportedPlatformData GetSupportedPlatforms(CompletionItem item, W
             return null;
         }
 
-        public static int GetContextPosition(CompletionItem item)
+        public static async Task<int> GetContextPositionAsync(
+            Document document, CompletionItem item, CancellationToken cancellationToken)
         {
-            if (item.Properties.TryGetValue("ContextPosition", out var text) && int.TryParse(text, out var number))
+            if (item.Properties.TryGetValue("ContextPosition", out var text) &&
+                int.TryParse(text, out var number))
             {
-                return number;
+                // We have no access to the editor at this layer.  So it's not 
```

Comment:
> Doesn't the rest of the completion system do the mapping before it gets into the providers to ensure this sort of a problem doesn't happen? 

No.  The entire abstraction is "CompletionService.GetDescription(Document, CompletionItem)".  

At this layer there is no concept of mapping.  One thing we could attempt is to have the higher layer of completion capture and hold onto the initial Document when the items were created.  We'd then pass that in instead of capturing the current document and passing that guy in instead.